go/consensus/tendermint/db/badger/badger.go

```diff
@@ -338,22 +338,22 @@ func (it *badgerDBIterator) Next() {
 		panic("Next with invalid iterator")
 	}
 
-	for {
```

Comment:
Wow, why did we keep scanning before? This looks like it kept scanning until reaching the end of the database because once the key was no longer inside the domain, we just kept going until the underlying iterator was valid.

Looking at the old bolt implementation it seems like the same was done there. Where did we get those semantics from, looking at other TM-DB backends, this doesn't seem to be the case? cc @Yawning

Reply:
I don't remember, sorry.

Reply:
key suspect is this https://github.com/oasislabs/oasis-core/blob/v20.1.2/go/consensus/tendermint/db/badger/badger.go#L302

I don't yet understand the case where we would need that, and I want to try to write an implementation that doesn't rely on that. But if it's true that we need to scan into the beginning of the range, that's why we'd need this loop.